Feed update time is max(newest post time, feed update time)

eldipa opened the pull request:
Some RSS sources (like Youtube) don't update the feed update time
(feed.feed.updated). It is a valid date but it is not updated when a new
post is added.

To overcome this we get the update time of the newest post and compare
it with the feed update time. We use the newest of both (max).

Closes #10

eldipa committed:
Feed update time is max(newest post time, feed update time)

Some RSS sources (like Youtube) don't update the feed update time
(feed.feed.updated). It is a valid date but it is not updated when a new
post is added.

To overcome this we get the update time of the newest post and compare
it with the feed update time. We use the newest of both (max).

sulami commented on feed2maildir/converter.py:
```diff
@@ -171,6 +168,23 @@ def find_update_time(self, feed):
+        # find the newest post and get its time
+        newest_post_time = self.find_update_time(feed)
+
+        try: # to get the update time from the feed itself
+            feed_time = self.mktime(feed.feed.updated)
+        except:
+            feed_time = None
+
+        if feed_time is None:
+            return newest_post_time
+        else:
+            # some feeds like Youtube does not update the feed's update time.
+            # the value 'feed_time' is a valid date but outdated so
+            # we have to compare it with the posts' times and get the latest.
+            return max(newest_post_time, feed_time)
```
This is a bit more concise:

```suggestion
        try: # to get the update time from the feed itself
            feed_time = self.mktime(feed.feed.updated)
        except:
            return newest_post_time
        # Some feeds like Youtube do not update the feed's update time.
        # The value 'feed_time' is a valid date, but outdated so
        # we have to compare it with the posts' times and get the latest.
        return max(newest_post_time, feed_time)
```

eldipa commented:
I like it. Updated the PR.

On Tue, Dec 15, 2020 at 02:20:42AM -0800, Robin Schroer wrote:
>@sulami commented on this pull request.
>
>
>
>> +        try: # to get the update time from the feed itself
>+            feed_time = self.mktime(feed.feed.updated)
>+        except:
>+            feed_time = None
>+
>+        if feed_time is None:
>+            return newest_post_time
>+        else:
>+            # some feeds like Youtube does not update the feed's update time.
>+            # the value 'feed_time' is a valid date but outdated so
>+            # we have to compare it with the posts' times and get the latest.
>+            return max(newest_post_time, feed_time)
>
>This is a bit more concise:
>
>```suggestion
>        try: # to get the update time from the feed itself
>            feed_time = self.mktime(feed.feed.updated)
>        except:
>            return newest_post_time
>        # some feeds like Youtube does not update the feed's update time.
>        # the value 'feed_time' is a valid date but outdated so
>        # we have to compare it with the posts' times and get the latest.
>        return max(newest_post_time, feed_time)
>```
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/sulami/feed2maildir/pull/11#pullrequestreview-552301669
